src/hivpy/hiv_status.py

```diff
@@ -24,17 +39,54 @@ def initial_HIV_status(self, population: pd.DataFrame):
         hiv_status = self._prob_HIV_initial(population[col.AGE]) > rng.random(len(population))
         return hiv_status
 
-    def update_HIV_status(self, population: pd.DataFrame):
+    def update_partner_risk_vectors(self, population):
+        """calculate the risk factor associated with each sex and age group"""
+        # Should we be using for loops here or can we do better? 
+        for sex in SexType:
+            for age_group in range(5):   # FIXME need to get number of age groups from somewhere
+                sub_pop_indices = population.data.index[(population.data[col.SEX]==sex) & (population.data[col.SEX_MIX_AGE_GROUP] == age_group)]
+                sub_pop = population.data.loc[sub_pop_indices]
+                n_stp_total = sum(sub_pop[col.NUM_PARTNERS])  # total number of people partnered to people in this group
+                n_infected = sum(sub_pop.loc[sub_pop[col.HIV_STATUS]][col.NUM_PARTNERS])  # num people parters to HIV+ people in this group
+                # Probability of being HIV prositive
+                if n_infected == 0:
+                    self.stp_HIV_rate[sex][age_group] = 0
+                else:
+                    self.stp_HIV_rate[sex][age_group] = n_infected/n_stp_total  # TODO: need to double check this definition              
+                # Chances of being in a given viral group
+                if n_stp_total > 0:
+                    self.stp_viral_group_rate[sex][age_group] = [sum(sub_pop.loc[sub_pop[col.VIRAL_LOAD_GROUP] == vg, col.NUM_PARTNERS])/n_stp_total for vg in range(7)]
+                else:
+                    self.stp_viral_group_rate[sex][age_group] = np.array([1,0,0,0,0,0,0])
+
+
+    def set_dummy_viral_load(self, population):
+        """Dummy function to set viral load until this part of the code has been implemented properly"""
+        population.data[col.VIRAL_LOAD_GROUP] = rng.choice(7, population.size)
+
+    def stp_HIV_transmission(self, person):
+        # TODO: Add circumcision, STIs etc. 
+        """Returns True if HIV transmission occurs, and False otherwise"""
+        stp_viral_groups = np.array([rng.choice(7, p=self.stp_viral_group_rate[1 - person[col.SEX]][age_group]) for age_group in person[col.STP_AGE_GROUPS]])
+        HIV_probabilities = np.array([self.stp_HIV_rate[1 - person[col.SEX]][age_group] for age_group in person[col.STP_AGE_GROUPS]])
+        viral_transmission_probabilities = np.array([max(0, rng.normal(self.transmission_means[group], self.transmission_sigmas[group])) for group in stp_viral_groups])
+        if person[col.SEX] == SexType.Female:
```

Comment:
Preferred per [the docs on enums](https://docs.python.org/3/library/enum.html#comparisons):
```suggestion
        if person[col.SEX] is SexType.Female:
```